robot_modeling/DQ_SerialManipulatorDH.m

```diff
@@ -53,6 +53,7 @@
 
```

Comment:
Again, I don't understand your credit attribution here. As far as I know, @mmmarinho wasn't the main contributor to this file. If I recall correctly (and I really don't want to waste my time going to the old repository on Sourceforge to double-check a code that I uploaded in 2011), this is something that I implemented in 2010! As we clearly discussed before, just reorganization of files does not contribute exactly a contribution to that file.

Maybe you are referring to some optimizations here and there that @mmmarinho made? If yes, I'm happy to see his name included in the contributions, but I don't understand why my name has been removed. Please clarify.

 (@mmmarinho, please correct me if I'm wrong).

Reply:
> Again, I don't understand your credit attribution here. As far as I know, @mmmarinho wasn't the main contributor to this file. If I recall correctly (and I really don't want to waste my time going to the old repository on Sourceforge to double-check a code that I uploaded in 2011), this is something that I implemented in 2010! As we clearly discussed before, just reorganization of files does not contribute exactly a contribution to that file.
> 
> Maybe you are referring to some optimizations here and there that @mmmarinho made? If yes, I'm happy to see his name included in the contributions, but I don't understand why my name has been removed. Please clarify.
> 
> (@mmmarinho, please correct me if I'm wrong).

Hi @bvadorno, 

I did not attribute credits in that file. I only updated the year and the documentation.

Reply:
So, this seems like a legacy mistake that I didn't spot and must be fixed. @mmmarinho, feel free to jump into this conversation and shed some light on what's happening. I must confess I become **_really_** confused when those things happen.

Reply:
@bvadorno I think this is a good opportunity to continuously refine how credit is attributed in the source files.

This particular file did not exist before, and is basically the only reason why @bvadorno was not listed. For example, `DQ.cpp` etc were initially basically a copy-and-paste of the MATLAB version, but for the longest time it didn't have any attribution to @bvadorno.

As far as credit goes, of course @bvadorno must be attributed and in this regard it was my mistake. But as far as accountability goes, @bvadorno should not be held accountable for a source file they did not write. So I guess the point is, where do we draw the line here? 

Depending on the answers to this, I think we have to revise most of the source files in most languages as well. 

One possible approach is to credit the original paper on the documentation of specific methods (so that they receive credit) but they are not liable for a file they did not write.

Reply:
Hello, @bvadorno,

I think that is a great compromise between receiving credit and having accountability.

So, to sum up:
1. Reference the original paper on the functions/methods and/or classes as required.
2. Add a more descriptive acknowledgment at the 'contribution' part of the header of the file.

I think it might take a while to backtrack and fix all files, but can we start doing that in the new ones (and newly modified ones), @juanjqo? 

For `DQ_SerialManipulatorDH` maybe we can start with what @bvadorno proposed above. Here is a quick draft I made, but please @juanjqo  check the links if you can because I'm not too sure of the info. Anyways, just a draft for us to start moving.

> 1. @bvadorno  ([original_author@mail.com](mailto:original_author@mail.com))
> 
> Responsible for the original implementation in file `SerialManipulator.m` (link to the file, hopefully the original one at the GitHub history not the current one because precedence can be broken).
>
> 2. @mmmarinho  ([person_who_reorganized_the_files@mail.com](mailto:person_who_reorganized_the_files@mail.com))
> 
> Created this file by reorganizing the code in the original file to comply with `SerialManipulator.m` becoming an abstract class, according to the discussion at https://github.com/dqrobotics/matlab/pull/56. Added support for prismatic joints. (https://github.com/dqrobotics/matlab/commit/f5aa70ac6a0a676557543e2bf7c418ab05c47326)
>
> 3. @juanjqo  ([third_person@mail.com](mailto:third_person@mail.com))
> 
> Did x and y according to the discussion at https://github.com/dqrobotics/matlab/pull/75.

Then maybe we can start discussing the details after seeing a first proposal. How about this, @bvadorno ?

Kind regards,
Murilo

Reply:
Hi, @mmmarinho,

I think it's great! 

Best wishes,
Bruno

Reply:
Hi @mmmarinho, 

I'm going to update the attribution of the credits in the new files (and newly modified ones), as you suggested. 

Best regards, 

Juancho

Reply:
@dqrobotics/developers , 

I added the following in the header of the DQ_SerialManipulatorDH. @bvadorno and @mmmarinho, what do you think?

```matlab
% Contributors to this file:
%     1. Bruno Vihena Adorno (adorno@ieee.org)
%        Responsible for the original implementation in file SerialManipulator.m 
%        [bvadorno committed on Apr 10, 2019] (bc7a95f)
%        (https://github.com/dqrobotics/matlab/blob/bc7a95f064b15046f43421d418946f60b1b33058/robot_modeling/DQ_SerialManipulator.m).
%
%     2. Murilo M. Marinho (murilo@nml.t.u-tokyo.ac.jp)
%        - Created this file by reorganizing the code in the original file to comply 
%          with SerialManipulator.m becoming an abstract class, according to the discussion 
%          at #56 (https://github.com/dqrobotics/matlab/pull/56).
% 
%        - Added support for prismatic joints. 
%          [mmmarinho committed on Apr 28, 2020] (f5aa70a) 
%          https://github.com/dqrobotics/matlab/commit/f5aa70ac6a0a676557543e2bf7c418ab05c47326
%
%     3. Juan Jose Quiroz Omana (juanjqo@g.ecc.u-tokyo.ac.jp)
%        - Added some modifications discussed at #56 (https://github.com/dqrobotics/matlab/pull/75)
%          to define DQ_SerialManipulator as an abstract class.
```

Best regards, 

Juancho

Reply:
Hi @juanjqo, 

I think it's sufficient for proper credit attribution. The implementation inside `DQ_SerialManipulator` predates the class itself and was done one decade ago. However, we don't need to go as far as that. 😃 

Best wishes,
Bruno

Reply:
Hi @bvadorno, 

Yes, you are right. I tried to trace the first commit where the DQ_SerialManipulator. However, that implementation is older!